### PR TITLE
Local Activity vs Activity

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

recording our internal Local Activity vs Activity discussion. just bumping up the quality of our doc on this and linking to the detailed forum post. cc @macrogreg

### CODE REVIEW

File: docs/content/what-is-a-local-activity.md [hunk 1]
```diff
@@ -6,13 +6,19 @@ description: todo
 
 A Local Activity is an [Activity Execution](/docs/content/what-is-an-activity-execution) that executes in the same process as the [Workflow Execution](/docs/content/what-is-a-workflow-execution) that spawns it.
 
-Although a Local Activity consumes less resources than a regular Activity Execution, it is subject to shorter durations and a lack of rate limiting.
+Some Activities are very short lived and do not need the queuing semantic, flow control, rate limiting and routing capabilities.
+For this case, Temporal supports a _local Activity_ feature.
 
-Some Activities are very short lived and do not need the queuing semantic, flow control, rate limiting and routing capabilities. For this case, Temporal supports a _local Activity_ feature. Local Activities are executed in the same worker process as the Workflow that invoked them. Consider using local Activities for functions that are:
+The main benefit of local Activities is that they are much more efficient in utilizing Temporal service resources and have much lower latency overhead compared to the usual Activity invocation.
```
Comment (flossypurse) [inline docs/content/what-is-a-local-activity.md:12]:
```suggestion
The main benefit of local Activities is that they are much more efficient in utilizing Temporal service resources and have much lower latency overhead compared to the usual Activity Executions.
```

What does it mean to be "more efficient in utilizing Temporal service resources"? I actually don't know what that means here - any way we can be explicit?

Same thing with "latency overhead" - are we specifically referring to the fact that the Workflow Execution does not have to wait like a normal Activity Execution?

File: docs/content/what-is-a-local-activity.md [hunk 2]
```diff
@@ -6,13 +6,19 @@ description: todo
...
+Consider using local Activities for functions that are:
 
-- no longer than a few seconds
+- no longer than a few seconds, inclusive of retries (shorter than the Workflow Task Timeout, which is 10 seconds by default)
 - do not require global rate limiting
 - do not require routing to specific workers or pools of workers
 - can be implemented in the same binary as the Workflow that invokes them
 
-The main benefit of local Activities is that they are much more efficient in utilizing Temporal service resources and have much lower latency overhead compared to the usual Activity invocation.
+Using local activities without understanding their limitations will cause various production issues.
```
Comment (flossypurse) [inline docs/content/what-is-a-local-activity.md:22]:
```suggestion
Using a Local Activity without understanding its limitations can cause various production issues.
```

File: docs/content/what-is-a-local-activity.md [hunk 3]
```diff
@@ -6,13 +6,19 @@ description: todo
...
+**Our advice is to stick to normal activities unless your use case requires very high throughput and large activity fanouts of very short activities.**
```
Comment (flossypurse) [inline docs/content/what-is-a-local-activity.md:23]:
```suggestion
**We recommend using regular Activities unless your use case requires very high throughput and large Activity fan outs of very short lived Activities.**
```

**Comment by flossypurse** [comment]
Can merge master to build successfully I think.

File: docs/content/what-is-a-local-activity.md [hunk 4]
```diff
@@ -6,13 +6,19 @@ description: todo
 
 A Local Activity is an [Activity Execution](/docs/content/what-is-an-activity-execution) that executes in the same process as the [Workflow Execution](/docs/content/what-is-a-workflow-execution) that spawns it.
 
-Although a Local Activity consumes less resources than a regular Activity Execution, it is subject to shorter durations and a lack of rate limiting.
+Some Activities are very short lived and do not need the queuing semantic, flow control, rate limiting and routing capabilities.
```
Comment (djmagee) [inline docs/content/what-is-a-local-activity.md:9]:
```suggestion
Some Activities are very short lived and do not need the queuing semantic, flow control, rate limiting and routing capabilities.
```
```suggestion
Some Activities are very short lived and do not need the queuing semantic, flow control, rate limiting, and routing capabilities.
```

File: docs/content/what-is-a-local-activity.md [hunk 5]
```diff
@@ -6,13 +6,19 @@ description: todo
...
+For this case, Temporal supports a _local Activity_ feature.
```
Comment (djmagee) [inline docs/content/what-is-a-local-activity.md:10]: Here and in next sentence: Should we standardize on init-capping "Local Activity", @flossypurse?

File: docs/content/what-is-a-local-activity.md (hunk 1 above)
Comment (sw-yx) [reply docs/content/what-is-a-local-activity.md:12]: honestly i dont love that text either, i just copy pasted it from what we had before

File: docs/content/what-is-a-local-activity.md (hunk 5 above)
Comment (sw-yx) [reply docs/content/what-is-a-local-activity.md:10]:
```suggestion
For this case, Temporal supports a _Local Activity_ feature.
```

**Review by djmagee — APPROVED** [review]

**Pull request merged**